### PR TITLE
[KEYCLOAK-9489] - User not able to log in to admin console when using query-* roles

### PR DESCRIPTION
**Pull request by pedroigor**

### CODE REVIEW

**Review by hmlnarik — COMMENTED** [review]
This looks good to me but would like to have QE to have their say.

**Comment by pdrozd** [comment]
@keycloak-ci-bot test

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/PermissionsTest.java [hunk 1]
```diff
@@ -295,12 +295,12 @@ public void invoke(RealmResource realm) {
...
         assertGettersEmpty(clients.get(AdminRoles.QUERY_REALMS).realm(REALM_NAME).toRepresentation());
 
-        // this should throw forbidden as "query-users" role isn't enough
+        // this should pass given that users granted with "query" roles are allowed to access the realm with limited access
         invoke(new Invocation() {
             public void invoke(RealmResource realm) {
                 clients.get(AdminRoles.QUERY_USERS).realm(REALM_NAME).toRepresentation();
             }
-        }, clients.get(AdminRoles.QUERY_USERS), false);
+        }, clients.get(AdminRoles.QUERY_USERS), true);
```
Comment (mhajas) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/PermissionsTest.java]: I think we need similar test for each of query-* role. Something like this:

````
for (String role : AdminRoles.ALL_QUERY_ROLES) {
    invoke(realm -> clients.get(role).realms().realm(REALM_NAME).toRepresentation(), clients.get(role), true);
}
````

**Review by mhajas — CHANGES_REQUESTED** [review]
Great work @pedroigor. I just left you one comment.

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/PermissionsTest.java (hunk 1 above)
Comment (pedroigor) [reply testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/PermissionsTest.java]: Good point. I'll update the PR with your suggestion.

**Comment by pdrozd** [comment]
@keycloak-ci-bot test

**Review by mhajas — APPROVED** [review]

**Review by vmuzikar — APPROVED** [review]
This PR as such LGTM.

On a side note, this issue is imho closely related to [KEYCLOAK-9177](https://issues.jboss.org/browse/KEYCLOAK-9177). Even though, this PR fixes the query-* roles and users are now able to log in, they still cannot actually modify or even view any clients/users/whatever they should unless they have the `view-realm` role.
The point is, the `view-realm` role itself allows user to log in to the admin console. And users currently must have this role in order to view/modify just about  anything. So we wouldn't need the query-* roles to allow users to login to the admin console, in other words we wouldn't need this fix, if users must have `view-realm` role anyway.

**Review by pdrozd — APPROVED** [review]

**Comment by pedroigor** [comment]
Yeah, it is related. Thanks for pointing that out. The problem here is that fine-grained admin relies on the query roles to be able to allow users to log in and manage specifics parts of the realm.

However, you have a point here. The changes here are not enough to get all the functionality back. There are other changes introduced by KEYCLOAK-8166 that need to be revisited in order to not only allow users to log in but perform actions solely based on the permissions.

I'll review this and update the PR.

**Pull request merged**